ouroboros-consensus/src/Ouroboros/Consensus/Node/Run.hs

```diff
@@ -98,4 +111,16 @@ class ( LedgerSupportsProtocol           blk
       , ShowProxy                (Header blk)
       , ShowProxy                 (Query blk)
       , ShowProxy           (TxId (GenTx blk))
-      ) => RunNode blk
+      ) => RunNode blk where
+  -- | A protocol may or may not be able to determine from the 'Header' whether
+  -- the header's block was self-issued.
+  --
+  -- TODO Only the @Shelley@ currently returns anything other than
+  -- 'MaybeSelfIssued'.
+  isSelfIssued :: TopLevelConfig blk -> Header blk -> WhetherSelfIssued
+
+  default isSelfIssued ::
```

Comment:
I'm not keen on `default`, but it at least helps keep the diff small for this stage of the PR review.

Reply:
It actually confused us while we were reviewing this small PR ("wait, why does it build without an HFC instance?"). Definitely not going to be explicit!